media-libs/opencv/opencv-4.10.0.ebuild

```diff
@@ -0,0 +1,1061 @@
+			# due to https://github.com/opencv/opencv/issues/25920
+			if [[ -n "${CUDA_ARCH_BIN}" ]]; then
+				export CUDAARCHS="${CUDA_ARCH_BIN//./}"
+			else
+				export CUDAARCHS="$(cuda_get_host_native_arch)"
+			fi
+		fi
+
+		einfo "CUDAHOSTCXX: ${CUDAHOSTCXX}"
+		einfo "CUDAARCHS: ${CUDAARCHS}"
+		einfo "CUDA_ARCH_BIN: ${CUDA_ARCH_BIN:=${CUDAARCHS}}"
+		einfo "CUDA_ARCH_PTX: ${CUDA_ARCH_PTX:=${CUDAARCHS}}"
+		einfo "CUDA_GENERATION: ${CUDA_GENERATION}"
+
+		# WARNING: there is a bug that makes CMake to igore
+		# CUDA_ARCH_BIN/PTX and CUDA_GENERATION options in case
+		# the ENABLE_CUDA_FIRST_CLASS_LANGUAGE is set to "yes"
+		# https://github.com/opencv/opencv/issues/25920
+		# So the configuration below is broken at the moment.
+		# Please rely on the CUDAARCHS env variable for now!
+
+		mycmakeargs+=(
+			-DENABLE_CUDA_FIRST_CLASS_LANGUAGE="yes"
+			-DCUDA_ARCH_BIN="${CUDA_ARCH_BIN}"
+			-DCUDA_ARCH_PTX="${CUDA_ARCH_PTX}"
+		)
+
+		if [[ -n "${CUDA_GENERATION}" ]]; then
+			mycmakeargs+=(
+				-DCUDA_GENERATION="${CUDA_GENERATION}"
+			)
+		fi
+	fi
+
+	if use ffmpeg; then
+		mycmakeargs+=(
+			-DOPENCV_GAPI_GSTREAMER="no"
+		)
+	fi
+
+	if use mkl; then
+		mycmakeargs+=(
+			-DLAPACK_IMPL="MKL"
+			-DMKL_WITH_OPENMP="$(usex !tbb "$(usex openmp)")"
+			-DMKL_WITH_TBB="$(usex tbb)"
+		)
+	fi
+
+	# NOTE set this via MYCMAKEARGS if needed
+	if use opencl; then
+		if has_version sci-libs/clfft; then
+			mycmakeargs+=( -DWITH_OPENCLAMDFFT="yes" )
+		else
+			mycmakeargs+=( -DWITH_OPENCLAMDFFT="no" )
+		fi
+		if has_version sci-libs/clblas; then
+			mycmakeargs+=( -DWITH_OPENCLAMDBLAS="yes" )
+		else
+			mycmakeargs+=( -DWITH_OPENCLAMDBLAS="no" )
+		fi
+	else
+		mycmakeargs+=(
+			-DWITH_OPENCLAMDFFT="no"
+			-DWITH_OPENCLAMDBLAS="no"
+		)
+	fi
+
+	if use test; then
+		# opencv tests assume to be build in Release mode
+		CMAKE_BUILD_TYPE="Release"
+		mycmakeargs+=(
+			-DOPENCV_TEST_DATA_PATH="${WORKDIR}/${PN}_extra-${PV}/testdata"
+		)
+		if use vtk; then
+			mycmakeargs+=(
+				-DVTK_MPI_NUMPROCS="$(nproc)" # TODO
+			)
+		fi
+	fi
+
+	if multilib_is_native_abi && use python; then
+		python_configure() {
+			# Set all python variables to load the correct Gentoo paths
+			local mycmakeargs=(
+				"${mycmakeargs[@]}"
+				# python_setup alters PATH and sets this as wrapper
+				# to the correct interpreter we are building for
+				-DBUILD_opencv_python3="yes"
+				-DBUILD_opencv_python_bindings_generator="yes"
+				-DBUILD_opencv_python_tests="$(usex test)"
+				-DPYTHON_DEFAULT_EXECUTABLE="${EPYTHON}"
+				-DINSTALL_PYTHON_EXAMPLES="$(usex examples)"
+			)
+			cmake_src_configure
+		}
+
+		python_foreach_impl python_configure
+	else
+		mycmakeargs+=(
+			-DPYTHON_EXECUTABLE="no"
+			-DINSTALL_PYTHON_EXAMPLES="no"
+			-DBUILD_opencv_python3="no"
+			-DBUILD_opencv_python_bindings_generator="no"
+			-DBUILD_opencv_python_tests="no"
+		)
+		cmake_src_configure
+	fi
+	use java && java-ant-2_src_configure
```

Comment:
dropping `java-ant-2.eclass` means to also drop this line and instead call the `eant` function within `src_compile()`.
For example see [arduino-1.8.19-r1](https://github.com/gentoo/gentoo/blob/master/dev-embedded/arduino/arduino-1.8.19-r1.ebuild).

Reply:
Done in c0b97f4